toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -93,60 +103,76 @@ func CustomizeImage(buildDir string, baseConfigPath string, config *imagecustomi
 	}
 
 	// Convert image file to raw format, so that a kernel loop device can be used to make changes to the image.
-	buildImageFile := filepath.Join(buildDirAbs, BaseImageName)
+	rawImageFile := filepath.Join(buildDirAbs, BaseImageName)
+	defer func() {
+		cleanupErr := file.RemoveFileIfExists(rawImageFile)
+		if cleanupErr != nil {
+			if err != nil {
+				err = fmt.Errorf("%w:\nfailed to clean-up (%s): %w", err, rawImageFile, cleanupErr)
+			} else {
+				err = fmt.Errorf("failed to clean-up (%s): %w", rawImageFile, cleanupErr)
+			}
+		}
+	}()
 
-	logger.Log.Infof("Mounting base image: %s", buildImageFile)
-	err = shell.ExecuteLiveWithErr(1, "qemu-img", "convert", "-O", "raw", imageFile, buildImageFile)
+	logger.Log.Infof("Creating raw base image: %s", rawImageFile)
+	err = shell.ExecuteLiveWithErr(1, "qemu-img", "convert", "-O", "raw", imageFile, rawImageFile)
 	if err != nil {
 		return fmt.Errorf("failed to convert image file to raw format:\n%w", err)
 	}
 
 	// Customize the partitions.
-	partitionsCustomized, buildImageFile, err := customizePartitions(buildDirAbs, baseConfigPath, config, buildImageFile)
+	partitionsCustomized, rawImageFile, err := customizePartitions(buildDirAbs, baseConfigPath, config, rawImageFile)
 	if err != nil {
 		return err
 	}
 
 	// Customize the raw image file.
-	err = customizeImageHelper(buildDirAbs, baseConfigPath, config, buildImageFile, rpmsSources, useBaseImageRpmRepos,
+	err = customizeImageHelper(buildDirAbs, baseConfigPath, config, rawImageFile, rpmsSources, useBaseImageRpmRepos,
 		partitionsCustomized)
 	if err != nil {
 		return err
 	}
 
 	// Shrink the filesystems.
 	if enableShrinkFilesystems {
-		err = shrinkFilesystemsHelper(buildImageFile, outputImageFile)
+		err = shrinkFilesystemsHelper(rawImageFile)
 		if err != nil {
 			return fmt.Errorf("failed to shrink filesystems:\n%w", err)
 		}
 	}
 
 	if config.SystemConfig.Verity != nil {
 		// Customize image for dm-verity, setting up verity metadata and security features.
-		err = customizeVerityImageHelper(buildDirAbs, baseConfigPath, config, buildImageFile, rpmsSources, useBaseImageRpmRepos)
+		err = customizeVerityImageHelper(buildDirAbs, baseConfigPath, config, rawImageFile, rpmsSources, useBaseImageRpmRepos)
 		if err != nil {
 			return err
 		}
 	}
 
 	// Create final output image file if requested.
-	if outputImageFormat != "" {
+	switch outputImageFormat {
+	case ImageFormatVpc, ImageFormatVhdx, ImageFormatQCow2, ImageFormatRaw:
 		logger.Log.Infof("Writing: %s", outputImageFile)
 
 		outDir := filepath.Dir(outputImageFile)
```

Comment:
nit: can remove this and use outputImageDir

Reply:
fixed.